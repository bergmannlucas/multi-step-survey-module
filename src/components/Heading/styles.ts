import styled, { css, DefaultTheme } from 'styled-components'
import { HeadingProps } from '.'

const wrapperModifiers = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
  `,

  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.huge};
  `
}

export const Wrapper = styled.h2<HeadingProps>`
  ${({ theme, size }) => css`
    color: ${theme.colors.purple};

    font-weight: ${theme.font.bold};

    ${!!size && wrapperModifiers[size](theme)}
  `}
`
