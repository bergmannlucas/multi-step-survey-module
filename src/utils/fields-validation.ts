import Joi from 'joi'

export const schema = Joi.array().items(
  Joi.object({
    name: Joi.string().empty(''),
    email: Joi.string()
      .email({ tlds: { allow: false } })
      .empty('')
  }),
  Joi.object({
    gender: Joi.string().required(),
    age: Joi.number().required()
  }),
  Joi.object({
    book: Joi.string().required(),
    colors: Joi.object({
      black: Joi.boolean(),
      white: Joi.boolean(),
      red: Joi.boolean(),
      green: Joi.boolean(),
      blue: Joi.boolean(),
      yellow: Joi.boolean(),
      purple: Joi.boolean(),
      orange: Joi.boolean(),
      pink: Joi.boolean()
    })
  })
)
