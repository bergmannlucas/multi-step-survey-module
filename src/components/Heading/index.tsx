import React from 'react'

import * as S from './styles'

export type HeadingProps = {
  children: React.ReactNode
  size?: 'small' | 'huge'
}

function Heading({ size = 'huge', children }: HeadingProps) {
  return <S.Wrapper size={size}>{children}</S.Wrapper>
}

export default Heading
