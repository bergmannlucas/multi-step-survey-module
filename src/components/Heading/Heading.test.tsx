import { render, screen } from 'utils/test-utils'
import Heading from '.'

describe('<Heading />', () => {
  it('should render a purple heading by default', () => {
    const { container } = render(<Heading>Identity</Heading>)
    expect(screen.getByRole('heading', { name: /identity/i })).toHaveStyle({
      color: '#9249FF'
    })
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a heading with a small size', () => {
    render(<Heading size="small">Identity</Heading>)

    expect(screen.getByRole('heading', { name: /identity/i })).toHaveStyle({
      'font-size': '1.6rem'
    })
  })

  it('should render a heading with a huge size', () => {
    render(<Heading size="huge">Identity</Heading>)

    expect(screen.getByRole('heading', { name: /identity/i })).toHaveStyle({
      'font-size': '5.2rem'
    })
  })
})
