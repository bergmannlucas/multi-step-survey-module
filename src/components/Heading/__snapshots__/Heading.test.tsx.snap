// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render a purple heading by default 1`] = `
.c0 {
  color: #9249FF;
  font-weight: 700;
  font-size: 5.2rem;
}

<h2
  class="c0"
>
  Identity
</h2>
`;
